src/Diagnostics.ModelsAndUtils/Models/ResponseExtensions/Solution.cs

```diff
@@ -8,123 +8,96 @@ namespace Diagnostics.ModelsAndUtils.Models.ResponseExtensions
 {
     public class Solution
     {
+        /// <summary>
+        /// Name of the Solution.
+        /// </summary>
+        public string Name { get; set; }
+
         /// <summary>
         /// Solution title.
         /// </summary>
         public string Title { get; set; }
 
         /// <summary>
-        /// Descriptions which will be rendered as Markdown.
+        /// A description which will be rendered as Markdown.
         /// </summary>
         public string Description { get; set; }
 
         /// <summary>
-        /// Name of the action for display customization purposes.
+        /// Denotes which action will be performed, such as calling an ARM API or navigating to a Portal Blade.
         /// </summary>
-        public string ActionName { get; set; }
+        [JsonConverter(typeof(StringEnumConverter))]
+        public ActionType Action { get; set; }
 
         /// <summary>
-        /// If the solution requires confirmation, it will signify a potentially dangerous action in the UI.
+        /// Options pertaining to <see cref="ActionType.ArmApi"/>.
+        /// This is ignored if <see cref="Action"/> is not <see cref="ActionType.ArmApi"/>.
         /// </summary>
-        public bool RequiresConfirmation { get; set; }
+        public ArmApiOptions ApiOptions { get; set; }
```

Comment:
>ArmApiOptions [](start = 15, length = 13)

Do you think it is better to have just one Options class  as base class and have all these ArmOptions, BladeOptions, TabOptions derive from it? That way based on ActionType the right options is loaded. It also becomes easier for detector author to just pass Options and they wont have to remember each type.

Reply:
I don't think this fits into a polymorphic model, none of the objects share similar attributes or methods and they aren't acted on similarly.